scripts/create_ocd_ids.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import print_function
+import csv
+"""
+Program to generate OCD IDs for the 2019 Indian Vidhan Sabha elections.
+"""
+
+contests = {"hr": "Haryana", "mh": "Maharashtra"}
+columns = ["id", "name"]
+country = "in"
+election = "Vidhan Sabha"
+write_table = []
+new_file = True
+
+# replace for other countries
+district_replacements = {
+    "district": " ",
+    "Yamuna Nagar": "Yamunanagar",
+    "Gondia": "Gondiya",
+    "Gurugram": "Gurgaon",
+    "Mewat": "Nuh",
+    "Dhamangaon": "Dhamamgaon",
+    "Deolali": "Devlali",
+    "Chikhali": "Chikhli",
+    "Ballabgarh": "Ballabhgarh",
+    "Kalanwali": "Kalawali",
+    "Navapur": "Nawapur",
+    "Sangole": "Sangola",
+    "Yewla": "Yevla",
+    "Jat": "Jath",
+    "Kalvan": "Kalwan",
+    "Kasba Peth": "Kasbapeth"
+}
+
+const_replacements = {
+    "Yamuna Nagar": "Yamunanagar",
+    "Gondia": "Gondiya",
+    "Gurugram": "Gurgaon",
+    "Mewat": "Nuh",
+    "Dhamangaon": "Dhamamgaon",
+    "Deolali": "Devlali",
+    "Chikhali": "Chikhli",
+    "Ballabgarh": "Ballabhgarh",
+    "Charkhi Dadri": "Dadri",
+    "Kalanwali": "Kalawali",
+    "Navapur": "Nawapur",
+    "Sangole": "Sangola",
+    "Yewla": "Yevla",
+    "Jat": "Jath",
+    "Kalvan": "Kalwan",
+    "Kasba Peth": "Kasbapeth"
+}
+
+punc_replacements = {
+    " ": "_",
+    "(": "",
+    ")": "",
+    "-": "_"
+}
+
+def read_csv(csv_file):
+  # returns in-memory list of table rows
+  table = []
+  with open(csv_file, "rb") as f:
+      csv_reader = csv.DictReader(f)
+      for row in csv_reader:
+        table.append(row)
+  return table
+
+def join_table(consts, districts, state, state_abbr):
+  # return joined table
+  new_table = []
+  ht = {}
+  for d_row in districts:
+    if d_row["district"] in district_replacements:
+      district_key = district_replacements[d_row["district"]]
+    else:
+      district_key = d_row["district"]
+    ht[district_key] = d_row["abbreviation"].lower()
+
+  for c_row in consts:
+    # source of truth on district names:
+    # https://affidavit.eci.gov.in/
```

Comment:
What, exactly is being pulled from there? What's the input CSV that this script is munging?

Reply:
I will include another script I made that fetches data and creates constituency CSVs for each state from the new source. District and constituency information is pulled from that website. I'll detail the expected format of the district abbreviation in a comment, but that information must be retrieved from elsewhere; in this case, they were taken from wikipedia manually put into a spreadsheet without the use of any provided script.

Reply:
While that's useful to have, I'd prefer to split that out into a separate PR and have this one focus on only the OCD-IDs.

Reply:
Great. I split the PRs with this one containing the OCD IDs and [another](https://github.com/opencivicdata/ocd-division-ids/pull/177) that adds the scripts that generate them.